src/libostree/ostree-sepolicy.c

```diff
@@ -300,6 +303,12 @@ initable_init (GInitable     *initable,
         goto out;
       self->rootfs_dfd = self->rootfs_dfd_owned;
     }
+  else if (!self->path)
```

Comment:
Ahh, OK that makes sense. It took me a bit to grok the logic here because they're just checking for the other one. Since we're sure now from the property setter that we have a proper XOR, can we rewrite these as:

```c
if (self->path)
  {
    ...
  }
else if (self->rootfs_dfd != -1)
  {
    ...
  }
else
  g_assert_not_reached ();
```

?